### ISSUE TITLE
SAML openssl.cnf is missing / Invalid

### ISSUE THREAD

**Issue by MattyHumphrey** [grokability/snipe-it#9263]
#### Expected Behavior (or desired behavior if a feature request)

(what you expect to happen goes here)

Clicking save should then enable SAML. 

-----

#### Actual Behavior

(what actually happens goes here)

When I click save I get the error message openssl.cnf is missing / Invalid. 
I have checked and I do have the PHPseclib files within the vendor folder.

-----

#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade? Upgrade 
- Version of Snipe-IT you're running v5.1.1
- Version of PHP you're running7.4.12
- Version of MySQL/MariaDB you're running 10.5.7
- What OS and web server you're running Snipe-IT on Windows Server 2019
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) Manual Install
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error 
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error. SAML Page
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.
- Include any additional information you can find in `storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

https://snipe-it.readme.io/docs/getting-help

**Comment by snipe** [comment]
@uberbrady can you take a look?

**Comment by uberbrady** [comment]
Specifically, I think what it's looking for is something along the lines of `/etc/openssl.cnf`, though different distros put it in different places.

In my CentOS-derivative, it gets installed under `/etc/pki/tls/openssl.cnf`. The `openssl` RPM seems to be what installed that there for me.

Can you see if you can install a library like that to get that file installed for your particular flavor of Linux?

**Comment by MattyHumphrey** [comment]
I have an openssl.cnf file which is inside Vendor/Phpseclib/phpseclib/phpseclib 

# minimalist openssl.cnf file for use with phpseclib

HOME			= .
RANDFILE		= $ENV::HOME/.rnd

[ v3_ca ]

**Comment by aranar-pro** [comment]
Curious if this is resolved.  Windows Server 2019/IIS there is a default openssl.cnf located at PHP\extras\ssl and you have to change your windows system environment variable OPENSSL_CONF to wherever you choose to put it.  Is Snipe-IT failing to find the one included in Vendor or should environment variable just be mentioned in the [Docs](https://snipe-it.readme.io/docs/saml) for enabling SAML on WS2019/IIS/PHP?

**Comment by bryceprutsos** [comment]
> Curious if this is resolved. Windows Server 2019/IIS there is a default openssl.cnf located at PHP\extras\ssl and you have to change your windows system environment variable OPENSSL_CONF to wherever you choose to put it. Is Snipe-IT failing to find the one included in Vendor or should environment variable just be mentioned in the [Docs](https://snipe-it.readme.io/docs/saml) for enabling SAML on WS2019/IIS/PHP?

@aranar-pro  install openssl we did this through chocolatey but the manual install should work.

**Comment by mmusah480** [comment]
I also have this issue please is there any solution

**Comment by uberbrady** [comment]
Apparently the default path for the openssl.cnf file might be:

C:/Openssl/openssl.cnf

If you copy that minimal openssl.cnf file into that location, does it work?

**Comment by mmusah480** [comment]
> Apparently the default path for the openssl.cnf file might be:
> 
> C:/Openssl/openssl.cnf
> 
> If you copy that minimal openssl.cnf file into that location, does it work?

Still the same, not working

**Comment by mmusah480** [comment]
Please any help, I could it be permission restrictions.

**Comment by mmusah480** [comment]
What file path does snipe points to for the openssl.cnf

**Comment by adagioajanes** [comment]
Also running into this issue utilizing Azure App Service with a Windows backend.

Dug into this some more. I still don't have an answer. But I was able to locate the openssl.cnf file on my platform.
C:\Program Files (x86)\PHP\v7.4\extras\ssl\opensll.cnf

Now I just need to tell Snipe IT (or PHP, or OpenSSL) to look for the configuration file there, instead of wherever it is currently looking.

**Comment by adagioajanes** [comment]
Alright, with Azure App Service on Windows, I was able to resolve this.
I had to specify an environment variable for OPENSSL_CONF and set it to C:\Program Files (x86)\PHP\v7.4\extras\ssl\opensll.cnf. That fixed it after a restart.

I'm going to file a ticket with Azure and let them know they need to fix their default configuration. The default one does not point to a valid file, nor path.

**Comment by aranar-pro** [comment]
Yes, I apologize if that wasn't clear in my comment, but I found for Windows Server I did have to specify the environment variable, so @mmusah480 if your openssl.cnf is at C:\Openssl\openssl.cnf you need to add OPENSSL_CONF with Value C:\Openssl\openssl.cnf to your system environment variable.

**Comment by adagioajanes** [comment]
It might be a good idea to put a little bit of this troubleshooting info into the documentation. I think the application is working correctly. But the error still doesn't really tell us what needs to be fixed other than 'openssl.cnf'.

Obviously, the documentation can't cover every environment, nor situation. But, I think some hints on where to look would be good. Maybe just explaining that it isn't setup correctly by default on all systems, and that you may need to specify an environment variable pointing it the right way.

**Comment by mmusah480** [comment]
@adagioajanes and @aranar-pro thanks for the assistance, apparently  I had the user variables configured and not the system variables, I can't belive the solution was hiding In plane sight. Thanks once again.

**Comment by aranar-pro** [comment]
I have suggested an edit to the Snipe-IT documentation under Windows/IIS specific setup so that at least the error is searchable (it's really a PHP thing but this would have helped me too) https://www.php.net/manual/en/openssl.installation.php

<blockquote><img src="https://www.php.net/favicon.ico" width="48" align="right"><div><strong><a href="https://www.php.net/manual/en/openssl.installation.php">PHP: Installation - Manual</a></strong></div></blockquote>

**Comment by rasari03** [comment]
I am using Windows server 2019 IIS and I don't have this C:\Program Files (x86)\PHP\v7.4\extras\ssl\openssl.cnf

or C:\Openssl\openssl.cnf.

In my case I had C:\PHP\extras\ssl\openssl.cnf and I did this to fix 
1. Open run Box in Windows Server 2019
2. type sysdm.cpl
3. In Sytem Properties window , go to Advanced Tab and click on Environment Variables
4. Under System variables Click new and add Variable OPENSSL_CONF and Value as C:\PHP\extras\ssl\openssl.cnf
5. Important to Reboot Server , as mine did not work until I did this.
